packages/tools/devtools/devtools-example/src/widgets/TaskManagerWidget.tsx

```diff
@@ -0,0 +1,100 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import React from "react";
+// import { ISharedMap } from "@fluidframework/map";
+import { TaskManager } from "@fluidframework/task-manager";
+
+/**
+ * {@link TaskManagerWidget} input props.
+ */
+export interface TaskManagerWidgetProps {
+	taskManager: TaskManager;
+}
+
+/**
+ * Simple TaskManager Widget.
+ * Queues the request to a task by the order it was received.
+ */
+export function TaskManagerWidget(props: TaskManagerWidgetProps): React.ReactElement {
+	const { taskManager } = props;
+
+	const taskId = "taskmManagerWidget-Id";
+
+	const [assigned, setAssigned] = React.useState<boolean>(false);
+	const [queued, setQueued] = React.useState<boolean>(false);
+	const [subscribed, setSubscribed] = React.useState<boolean>(false);
+
+	React.useEffect(() => {
+		if (taskManager !== undefined) {
+			const updateState = (): void => {
+				setAssigned(taskManager.assigned(taskId));
+				setQueued(taskManager.queued(taskId));
+				setSubscribed(taskManager.subscribed(taskId));
+			};
+
+			taskManager.on("assigned", updateState);
+			taskManager.on("lost", updateState);
+			taskManager.on("completed", updateState);
+
+			taskManager.subscribeToTask(taskId);
+
+			return (): void => {
+				taskManager.off("assigned", updateState);
+				taskManager.off("lost", updateState);
+				taskManager.off("completed", updateState);
+			};
+		}
+	}, [taskManager]);
+
+	if (taskManager === undefined) return <div />;
+
+	const abandon = (): void => taskManager.abandon(taskId);
+	const volunteer = async (): Promise<boolean> => taskManager.volunteerForTask(taskId);
+	const subscribe = (): void => taskManager.subscribeToTask(taskId);
+	const complete = (): void => taskManager.complete(taskId);
+
+	return (
+		<div style={{ margin: "15px 0px 15px 0px" }}>
+			<div className="task-manager-header">
+				<strong>Task Manager Info</strong>
+			</div>
+
+			<div className="task-manager-body">
+				<div>
+					{assigned
+						? "This Client is currently: Task Assignee"
+						: "This Client is currently: Not Task Assignee"}
+				</div>
+				<div>Queued: {queued.toString()}</div>
+				<div>Assigned: {assigned.toString()}</div>
+				<div>Subscribed: {subscribed.toString()}</div>
+
+				<div className="task-manager-controls" style={{ margin: "7px 0px 7px 0px" }}>
+					<button disabled={!queued} onClick={abandon} className="debug-controls button">
+						Abandon
+					</button>
+					<button disabled={queued} onClick={volunteer} className="debug-controls button">
+						Volunteer
+					</button>
```

Comment:
For buttons that execute asynchronous tasks, we should have a loading state. I'm not sure if we do this in our app at all but we should take note and implement it in the future.
You could do something like
```
taskManager.volunteerForTask(taskId).then(() => {
// do something
setIsVolunteerForTaskLoading(false);
});

setIsVolunteerForTaskLoading(true);
```

Then in your button:
```
<button disabled={queued || isVolunteerForTaskLoading} onClick={volunteer} className="debug-controls button">
Volunteer
</button>
```